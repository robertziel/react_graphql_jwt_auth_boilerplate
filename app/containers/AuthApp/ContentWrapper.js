import styled from 'styled-components';

const ContentWrapper = styled.div`
  width: 100%;
  height: 100%;
`;

export default ContentWrapper;
