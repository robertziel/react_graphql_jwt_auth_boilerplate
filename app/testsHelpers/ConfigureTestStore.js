import { memoryHistory } from 'react-router-dom';
import configureStore from 'configureStore';
import StoreAccessor from 'containers/ApiConnector/StoreAccessor';

export default function ConfigureTestStore() {
  const confuguredStore = configureStore({}, memoryHistory);

  StoreAccessor.store = confuguredStore.store;

  this.store = confuguredStore.store;
  this.presistor = confuguredStore.presistor;
}
