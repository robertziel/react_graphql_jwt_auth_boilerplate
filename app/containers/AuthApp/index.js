/**
 *
 * AuthApp.js
 *
 * This component is the skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar)
 *
 */

import React from 'react';
import { Redirect, Switch, Route } from 'react-router-dom';

import Footer from 'components/Footer/index';
import SignInPage from 'containers/_authPages/SignInPage/Loadable';
import SignUpPage from 'containers/_authPages/SignUpPage/Loadable';
import ContentWrapper from './ContentWrapper';

function AuthApp() {
  return (
    <ContentWrapper>
      <Switch>
        <Route exact path="/login" component={SignInPage} />
        <Route exact path="/sign-up" component={SignUpPage} />
        <Route>
          <Redirect to="/login" />
        </Route>
      </Switch>
      <Footer />
    </ContentWrapper>
  );
}

export default AuthApp;
