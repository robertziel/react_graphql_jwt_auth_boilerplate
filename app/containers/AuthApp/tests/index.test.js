import React from 'react';

import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ConfigureTestStore from 'testsHelpers/ConfigureTestStore';

import AuthApp from '../index';

function shallowWrapper() {
  return shallow(<AuthApp store={store} />);
}

let store;
let wrapper;

beforeEach(() => {
  store = new ConfigureTestStore().store;
  wrapper = shallowWrapper();
});

describe('<AuthApp />', () => {
  it('should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
