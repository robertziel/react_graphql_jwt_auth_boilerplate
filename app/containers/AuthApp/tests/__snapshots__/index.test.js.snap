// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthApp /> should match snapshot 1`] = `
<ContentWrapper>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/login"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/sign-up"
    />
    <Route>
      <Redirect
        to="/sign-in"
      />
    </Route>
  </Switch>
  <Footer />
</ContentWrapper>
`;
